src/main/java/org/gitlab4j/api/models/Setting.java

```diff
@@ -217,485 +307,629 @@
+     */
+    PUSH_EVENT_ACTIVITIES_LIMIT(Integer.class),
 
     /**
-     * (If enabled, requires: recaptcha_private_key and recaptcha_site_key) Enable
-     * recaptcha.
+     * (<strong>If enabled, requires:</strong> {@link #RECAPTCHA_PRIVATE_KEY} and 
+     * {@link #RECAPTCHA_SITE_KEY}) Enable reCAPTCHA.
      */
     RECAPTCHA_ENABLED(Boolean.class),
 
-    /** required by: recaptcha_enabled Private key for recaptcha. */
+    /**
+     * required by: {@link #RECAPTCHA_ENABLED} Private key for reCAPTCHA.
+     */
     RECAPTCHA_PRIVATE_KEY(String.class),
 
-    /** required by: recaptcha_enabled Site key for recaptcha. */
+    /**
+     * required by: {@link #RECAPTCHA_ENABLED} Site key for reCAPTCHA.
+     */
     RECAPTCHA_SITE_KEY(String.class),
 
     /**
-     * GitLab will periodically run git fsck in all project and wiki repositories to
-     * look for silent disk corruption issues.
+     * Maximum push size (MB).
+     */
+    RECEIVE_MAX_INPUT_SIZE(Integer.class),
+
+    /**
+     * GitLab will periodically run git fsck in all project and wiki repositories to look for
+     * silent disk corruption issues.
      */
     REPOSITORY_CHECKS_ENABLED(Boolean.class),
 
-    /** (Premium) Size limit per repository (MB) */
+    /**
+     * (PREMIUM | SILVER) Size limit per repository (MB)
+     */
     REPOSITORY_SIZE_LIMIT(Integer.class),
-
+    
     /**
-     * A list of names of enabled storage paths, taken from gitlab.yml. New projects
-     * will be created in one of these stores, chosen at random.
+     * A list of names of enabled storage paths, taken from gitlab.yml. New projects will be
+     * created in one of these stores, chosen at random.
      */
     REPOSITORY_STORAGES(String[].class),
 
     /**
-     * (If enabled, requires: two_factor_grace_period) Require all users to set up
-     * Two-factor authentication.
+     * (<strong>If enabled, requires:</strong> {@link #TWO_FACTOR_GRACE_PERIOD}) Require all
+     * users to set up Two-factor authentication.
      */
     REQUIRE_TWO_FACTOR_AUTHENTICATION(Boolean.class),
 
     /**
-     * Selected levels cannot be used by non-admin users for groups, projects or
-     * snippets. Can take private, internal and public as a parameter. Default is
-     * null which means there is no restriction.
+     * Selected levels cannot be used by non-admin users for groups, projects or snippets. Can
+     * take private, internal and public as a parameter. Default is null which means there is
+     * no restriction.
      */
     RESTRICTED_VISIBILITY_LEVELS(String[].class),
 
     /**
-     * The minimum allowed bit length of an uploaded RSA key. Default is 0 (no
-     * restriction). -1 disables RSA keys.
+     * The minimum allowed bit length of an uploaded RSA key. Default is 0 (no restriction).
+     * -1 disables RSA keys.
      */
-    RSA_KEY_RESTRICTION(Integer.class),
+    RSA_KEY_RESTRICTION(Boolean.class),

```

Comment:
The documentation indicates that ```0``` or ```-1``` is returned, why change this to ```Boolean.class````

Reply:
Good catch. My bad.